### PR TITLE
Handle client sending new NodeKey 

### PR DESCRIPTION
**Pull request by juanfont**
This commit changes a bit how we handle the registration workflow. Hopefully solves #32.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -390,66 +428,43 @@ func (h *Headscale) getMapKeepAliveResponse(mKey wgcfg.Key, req tailcfg.MapReque
...
-		m.IPAddress = ip.String()
-		m.NamespaceID = pak.NamespaceID
-		m.AuthKeyID = uint(pak.ID)
-		m.RegisterMethod = "authKey"
-		m.Registered = true
-		db.Save(&m)
-
-		resp.MachineAuthorized = true
-		resp.User = *pak.Namespace.toUser()
+	pak, err := h.checkKeyValidity(req.Auth.AuthKey)
```
Comment (cure) [inline api.go:433]: Can we add a check before this line that req.Auth is not nil? Otherwise, if it ever were nil, we'd get a runtime panic.

**Review by cure — APPROVED** [review]
Just one small nit, looks good otherwise, thanks!

File: api.go (hunk 1 above)
Comment (juanfont) [reply api.go:433]: But req.Auth cannot be nil https://github.com/tailscale/tailscale/blob/main/tailcfg/tailcfg.go#L602

Comment (cure) [reply api.go:433]: OK that's fair, never mind then thanks.

**Pull request merged**